### DISCUSSION TITLE
Global Auditing

### DISCUSSION THREAD

**Discussion by IamMuhammadHasib** [owen-it/laravel-auditing#978] [Ideas]
Normal auditing for a model looks like
```
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;
use OwenIt\Auditing\Contracts\Auditable;

class User extends Model implements Auditable
{
    use \OwenIt\Auditing\Auditable;

    // ...
}
```

For global audit (to audit all my models) I created a `app/Models/BaseModel.php` file
```
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use OwenIt\Auditing\Contracts\Auditable as AuditableContract;
use OwenIt\Auditing\Auditable;

class BaseModel extends Model implements AuditableContract
{
    use Auditable;
}
```
And then in my models extending the BaseModel instead of Model
```
<?php

namespace App;

class User extends BaseModel
{
    // ...
}
```

**Can you make it a feature?**
Also with my method I need to edit all my models to extend BaseModel. It can be problematic for a high number of models. Moreover, some models are created inside packages (e.g. `saptie/laravel-permission`, `Auth`), which doesn't work this way. So, I had to do it manually as docs described, which may prone to package update! So, badly in need of a global auditing.

**Comment by willpower232** [comment]
I don't think it is possible to listen to model events without modifying the model so I doubt this is possible from this packages perspective. At the very least, it also expects all the relations and methods to be available from the model which requires the two lines added.

I know spatie/laravel-permission lets you specify the model so you can always extend their model and add the auditable parts yourself.

Unless there is a drastic laravel change, I doubt the two lines you are worried about being changed would ever change.

**Reply by erikn69** [reply]
It would also cause that if you try to use `disableAuditing` for a certain model, it would disable all models auditing that extend `BaseModel` (_It seems like I saw it before_)
